cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BaseRDDCollection.java

```diff
@@ -125,41 +127,41 @@ protected BaseRDDCollection(JavaSparkExecutionContext sec, FunctionCache.Factory
       String cacheStorageLevelString = sparkConf.get(Constants.SPARK_PIPELINE_CACHING_STORAGE_LEVEL, 
                                                      Constants.DEFAULT_CACHING_STORAGE_LEVEL);
       StorageLevel cacheStorageLevel = StorageLevel.fromString(cacheStorageLevelString);
-      return wrap(rdd.persist(cacheStorageLevel));
+      return wrap(getUnderlying().persist(cacheStorageLevel));
```

Comment:
Can you just make getRDD()? This getUnderlying is pretty confusing as "underlying" now can be RDD, SQLDataset and probably Dataframe/Dataset soon.

Reply:
Good suggestion, I will rename this method.

Reply:
Actually, change of mind: for realtime pipelines, the "Underlying" class is JavaDStream instead of JavaRDD, so I assume that was the initial intention of using the word "Underlying"

Reply:
That's why I recommend to do just separate getters instead of this guy. Underlying is very confusing

Reply:
Note that for your case we can start by introducing getRDD without killing getUnderlying. We can do a kill separately.

Reply:
@tivv do you mind if I leave this change outside of the scope of this PR? I will open a follow up PR with this suggestion.

Reply:
Sure, I just don't want for this getUnderlying to be widely introduced so that refactoring later would be easier

Reply:
now that there is a SQLCollection, can't this be changed back to rdd?